### PR TITLE
[4] Draft for db corrections of extension states protected, locked, enabled

### PR DESCRIPTION
**Pull request by ReLater**
# Draft! Don't merge.

See discussion Issue https://github.com/joomla/joomla-cms/issues/28527

Draft for cleanup of extension states
protected
locked
enabled.

Please make suggestions after checking these states.

### CODE REVIEW

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 1]
```diff
@@ -0,0 +1,454 @@
+-- Set components AS LOCKED (can not be disabled/can not be uninstalled) and unprotect them and ENABLE them.
+
+UPDATE `#__extensions`
+SET `locked` = 1, `protected` = 0, `enabled` = 1
+WHERE (`type` = 'component' AND `element` IN (
+	'com_wrapper',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:6]:
```suggestion
```

**Comment by richard67** [comment]
@ReLater When later doing the final PR, it will not need to create new update SQL files, you can just modify the existing 4.0.0-2020-03-25.sql files. This is because we are not beta yet and not support updates between alphas or nightly builds.

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+	'com_admin',
+	'com_banners',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:8]:
```suggestion
```

**Comment by ReLater** [comment]
> When later doing the final PR, it will not need to create new update SQL files, you can just modify the existing 4.0.0-2020-03-25.sql files.

OK.

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+
+-- 
+-- Set components AS PROTECTED (can not be uninstalled) and ENABLE them and remove LOCKED state.
+
+UPDATE `#__extensions`
+SET `protected` = 1, `enabled` = 1, `locked` = 0
+WHERE (`type` = 'component' AND `element` IN (
+	'com_actionlogs',
+	'com_admin',
+	'com_ajax',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:327]:
```suggestion
	'com_ajax',
	'com_banners',
```

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+	'com_media',
+	'com_menus',
+	'com_messages',
+	'com_modules',
+	'com_newsfeeds',
+	'com_plugins',
+	'com_templates',
+	'com_content',
+	'com_config',
+	'com_redirect',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:26]:
```suggestion
```

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+UPDATE `#__extensions`
+SET `locked` = 1, `protected` = 0, `enabled` = 1
+WHERE (`type` = 'component' AND `element` IN (
+	'com_wrapper',
+	'com_admin',
+	'com_banners',
+	'com_cache',
+	'com_categories',
+	'com_checkin',
+	'com_contact',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:12]:
```suggestion
```

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+UPDATE `#__extensions`
+SET `protected` = 1, `enabled` = 1, `locked` = 0
+WHERE (`type` = 'component' AND `element` IN (
+	'com_actionlogs',
+	'com_admin',
+	'com_ajax',
+	'com_cache',
+	'com_categories',
+	'com_checkin',
+	'com_config',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:331]:
```suggestion
	'com_contact',
```

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+	'com_content',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:332]:
```suggestion
```

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+	'com_messages',
+	'com_modules',
+	'com_newsfeeds',
+	'com_plugins',
+	'com_templates',
+	'com_content',
+	'com_config',
+	'com_redirect',
+	'com_users',
+	'com_finder',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:28]:
```suggestion
```

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+WHERE (`type` = 'component' AND `element` IN (
+	'com_actionlogs',
+	'com_admin',
+	'com_ajax',
+	'com_cache',
+	'com_categories',
+	'com_checkin',
+	'com_config',
+	'com_content',
+	'com_cpanel',
```
Comment (ReLater) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-04-03-DRAFT.sql:333]:
```suggestion
	'com_finder',
```

**Comment by ReLater** [comment]
Closed.
See comment https://github.com/joomla/joomla-cms/issues/28527#issuecomment-608695925
Let's see if I should reopen then here

**Pull request closed without merging**